effekt/shared/src/main/scala/effekt/Intelligence.scala

```diff
@@ -18,27 +18,36 @@ trait Intelligence {
     symbol: Symbol,
     header: String,
     signature: Option[String],
-    description: Option[String]
+    description: Option[String],
+    documentation: Option[Doc]
   ) {
     def fullDescription: String = {
       val sig = signature.map(sig => s"```effekt\n$sig\n```").getOrElse { "" }
       val desc = description.getOrElse("")
+      val doc = showDocumentation(documentation)
 
       s"""|#### $header
           |$sig
-          |$desc
+          |$desc$doc
           |""".stripMargin
     }
 
     def shortDescription: String = {
       val sig = signature.map(sig => s"```effekt\n$sig\n```").getOrElse { "" }
+      val doc = showDocumentation(documentation)
 
       s"""|#### $header
-          |$sig
+          |$sig$doc
           |""".stripMargin
     }
   }
 
+  def showDocumentation(documentation: Option[Doc]): String =
+    documentation.flatten
+      .map('\n' +: _)
+      .getOrElse("")
+      .replace("\\n", "\n")
```

Comment:
Why is this needed?

Reply:
Is .flatten.map f the same as .flatMap f?

Reply:
It seems we have:
https://github.com/effekt-lang/effekt/blob/6e8b2b515ebc45304b2d38ab161f9be0f7cd2e8d/effekt/shared/src/main/scala/effekt/source/Tree.scala#L109

But then, why do we pass an `Option[Doc]` (see other comment)?

Also, maybe instead of `.map(...).getOrElse(...)`, a `match` would be clearer?

Reply:
Sorry for the late response.

> Why is this needed?

We only want to add a newline delimiter if the symbol actually has a doc comment, therefore we map on it and then .getOrElse. We need to unescape the newlines in order to show them in VS code as literal newlines (we build the string with escaped newlines in `RecursiveDescent`)

> Is `.flatten.map f` the same as `.flatMap f`

No, as it turns out. Also, I have changed the types from `Option[Doc]` to `Doc`, so this is not required anymore (see other comment)

> Also, maybe instead of .map(...).getOrElse(...), a match would be clearer?

It's of course a stylistic choice, but I think it is clear enough as is.